File: Dataset/Leetcode/train/83/323.py
class Solution:
    def XXX(self, head: ListNode) -> ListNode:
        if head == None : return head
        have = set()
        have.add(head.val)
        q = head
        while q and q.next and q.next:
            if q.next.val in have : q.next = q.next.next
            # 删除下一个结点，直到下一个结点符合条件
            else : 
                have.add(q.next.val)
                q  = q.next
                # 移动到下一个结点
        return head

File: Dataset/Leetcode/test/48/323.cpp
class Solution {
public:
    void XXX(vector<vector<int>>& matrix) {
        int n = matrix.size();
        // 上下翻转
        for(int i = 0; i < n >> 1; ++i)
            swap(matrix[i], matrix[n - i - 1]); 
        // 镜像翻转
        for(int i = 0; i < n; ++i)
            for(int j = i; j < n; ++j)
                swap(matrix[i][j], matrix[j][i]);
        return;
    }
};

